app/queries/matched_for_notification.rb

```diff
@@ -7,70 +7,46 @@ def initialize(query_field: :tags, scope: SubscriberList)
   end
 
   # Filter out subscriber lists:
-  # - where the keys from the supplied hash:
-  #   - Have a superset of the keys in the specified query_field for subscriber_lists
-  #   - Have an overlap with  the keys in the specified query_field for or_joined_facet_subscriber_lists
-  # - AND
+  # - where the keys from the supplied hash are a superset of the keys in the specified query_field AND
   # - If the operator is 'any':
   #     the key values in the subscriber list have at least one corresponding match in the supplied hash.
   # - If the operator is 'all'
   #   all key values in the subscriber list have corresponding matches in the supplied hash.
+  # Note that this means that not all keys from the the supplied hash are required to be matched.
```

Comment:
Is this line needed? If it is, should it part of the description for `any`?

Reply:
I literally just reverted the PR (but without the migration being reverted).  I'll update the comment to make more sense.